src/App.tsx

```diff
@@ -1,61 +1,138 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+function findUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const preparedTodoList: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
+
+const getNewId = (array: { id: number }[]) => (
+  Math.max(...array.map(todo => todo.id)) + 1
+);
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [todos, setTodos] = useState(preparedTodoList);
+
+  const [isErrorOnUserSelect, setErrorOnUserSelect] = useState(false);
+  const [isErrorOnTitleInput, setErrorOnTitleInput] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setSelectedUserId(0);
+  };
+
+  const handleTitleChange = (event: React.FormEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setErrorOnTitleInput(false);
```

Comment:
Do you need to trigger this function if`isErrorOnTitleInput` has already been false ? You should add a check and trigger the change when`isErrorOnTitleInput` is true